File: 1833-maximum-ice-cream-bars/1833-maximum-ice-cream-bars.java
class Solution {
    public int maxIceCream(int[] costs, int coins) {
        Arrays.sort(costs);
        int sum=0;
        int count=0;
        for(int i:costs){
            sum+=i;
            if(sum<=coins){
                count++;
            }
        }
        return count;
    }
}

File: 1833-maximum-ice-cream-bars/1833-maximum-ice-cream-bars.py
class Solution:
    def maxIceCream(self, costs: List[int], coins: int) -> int:
        costs.sort()
        count = 0
        for price in costs:
            if(coins == 0):
                return count
            if(price <= coins):
                count += 1
                coins = coins - price
            
        return count